# Week 1b : Implementation of toy problems.

## Camel - Bananas problem is stated as follows.
A person has 3000 bananas and a camel. The person wants to transport the maximum number of bananas to a destination which is 1000 KMs away, using only the camel as a mode of transportation. The camel cannot carry more than 1000 bananas at a time and eats a banana every km it travels.

## To find
What is the maximum number of bananas that can be transferred to the destination using only camel (no other mode of transportation is allowed).


# Code
Solve the water Camel and Bananas problem

In [1]:
#Getting the input from the users

dist=int(input("Enter the distance to the market :")) #Distance
total=int(input("Enter the total bananas initially :")) #Total bananas
maxload=int(input("Enter the max load capacity of the camel :")) #Maximum load of the camel

Enter the distance to the market :1000
Enter the total bananas initially :3000
Enter the max load capacity of the camel :1000


In [2]:
#Initializing other parameters
eachkm=1 
totalmoved=0

#Midpoint if found for efficient computing
midpoints=total//dist


In [3]:
#Wrting the function to find the number of maximum bananas transfered to the destination
#The below code reproduces the mathematical function needed to compute the solution
'''
3000-5(distance)=2000
2000-3(distance)=1000
1000-distance=Maximum Bananas Left
'''

for x in range(midpoints,1,-1):
  trips=2*x-1
  movednow=((total-maxload)-total)/trips
  total=total-maxload
  totalmoved=totalmoved+movednow

bananas=round(maxload-abs(totalmoved))
print("Total remaining banans to the destination is ",bananas)
  


Total remaining banans to the destination is  467


## Results
The camel can transfer a maximum of 467 bananas to the destination